Doesn't compile on some 32-bit Linux platforms

For example on `i686-unknown-linux-musl`

```
error[E0308]: mismatched types
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/linux-keyutils-0.2.2/src/keyring.rs:31:13
   |
31 |             u64::from(create)
   |             ^^^^^^^^^^^^^^^^^ expected `u32`, found `u64`
   |
  ::: /root/.cargo/registry/src/github.com-1ecc6299db9ec823/linux-keyutils-0.2.2/src/ffi/mod.rs:13:9
   |
13 |         $crate::ffi::keyctl_impl($op, $a2, $a3, 0, 0)
   |         ------------------------ arguments to this function are incorrect
   |
note: function defined here
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/linux-keyutils-0.2.2/src/ffi/functions.rs:64:15
   |
64 | pub(crate) fn keyctl_impl(
   |               ^^^^^^^^^^^
...
67 |     arg3: libc::c_ulong,
   |     -------------------


error[E0277]: the trait bound `types::KeySerialId: From<u32>` is not satisfied
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/linux-keyutils-0.2.2/src/keyring.rs:33:10
   |
33 |         .try_into()
   |          ^^^^^^^^ the trait `From<u32>` is not implemented for `types::KeySerialId`
   |
   = note: required for `u32` to implement `Into<types::KeySerialId>`
   = note: required for `types::KeySerialId` to implement `TryFrom<u32>`
   = note: required for `u32` to implement `TryInto<types::KeySerialId>`


error[E0277]: the trait bound `types::KeySerialId: From<u32>` is not satisfied
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/linux-keyutils-0.2.2/src/keyring.rs:63:10
   |
63 |         .try_into()
   |          ^^^^^^^^ the trait `From<u32>` is not implemented for `types::KeySerialId`
   |
   = note: required for `u32` to implement `Into<types::KeySerialId>`
   = note: required for `types::KeySerialId` to implement `TryFrom<u32>`
   = note: required for `u32` to implement `TryInto<types::KeySerialId>`


error[E0277]: the trait bound `types::KeySerialId: From<u32>` is not satisfied
   --> /root/.cargo/registry/src/github.com-1ecc[629](https://github.com/PyO3/maturin/actions/runs/4268683584/jobs/7431347988#step:4:630)9db9ec823/linux-keyutils-0.2.2/src/keyring.rs:121:10
    |
121 |         .try_into()
    |          ^^^^^^^^ the trait `From<u32>` is not implemented for `types::KeySerialId`
    |
    = note: required for `u32` to implement `Into<types::KeySerialId>`
    = note: required for `types::KeySerialId` to implement `TryFrom<u32>`
    = note: required for `u32` to implement `TryInto<types::KeySerialId>`


error: aborting due to 4 previous errors


Some errors have detailed explanations: E0277, E0308.

For more information about an error, try `rustc --explain E0277`.
```

--- a/.github/workflows/checks.yml
+++ b/.github/workflows/checks.yml
@@ -67,7 +67,7 @@ jobs:
   #
   # Cannot run the unit tests until we can specify a seccomp profile for cross'
   # docker images. Since the default seccomp profile blocks keyctl calls.
-  libdbus-sys-linux:
+  cross-linux:
     runs-on: ubuntu-latest
     needs: install-cross
     steps:

--- a/.github/workflows/checks.yml
+++ b/.github/workflows/checks.yml
@@ -85,9 +85,7 @@ jobs:
     - run: rustup toolchain install ${{ matrix.channel }}
     - run: chmod +x /tmp/cross
     - name: Build
-      run: /tmp/cross build --package dbus --features vendored --target ${{ matrix.target }}
-    #- name: Run tests
-    #  run: /tmp/cross test --package libdbus-sys --features vendored --target ${{ matrix.target }}
+      run: /tmp/cross build --features vendored --target ${{ matrix.target }}
     strategy:
       fail-fast: false
       matrix:

--- a/.github/workflows/checks.yml
+++ b/.github/workflows/checks.yml
@@ -85,7 +85,7 @@ jobs:
     - run: rustup toolchain install ${{ matrix.channel }}
     - run: chmod +x /tmp/cross
     - name: Build
-      run: /tmp/cross build --features vendored --target ${{ matrix.target }}
+      run: /tmp/cross build --all-features --target ${{ matrix.target }}
     strategy:
       fail-fast: false
       matrix:

--- a/.github/workflows/checks.yml
+++ b/.github/workflows/checks.yml
@@ -10,6 +10,28 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
+  # Taken from rust-lang/libz-sys CI/CD example.
+  #
+  # This job downloads and stores `cross` as an artifact, so that it can be
+  # redownloaded across all of the jobs.
+  install-cross:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: XAMPPRocky/get-github-release@v1
+        id: cross
+        with:
+          owner: rust-embedded
+          repo: cross
+          matches: ${{ matrix.platform }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: actions/upload-artifact@v3
+        with:
+          name: cross-${{ matrix.platform }}
+          path: ${{ steps.cross.outputs.install_path }}
+    strategy:
+      matrix:
+        platform: [linux-musl]
+
   # Build and run unit tests
   build:
     runs-on: ubuntu-latest
@@ -40,6 +62,45 @@ jobs:
       run: cargo clippy --verbose -- --deny "warnings"
     - name: Run RustFmt
       run: cargo fmt -- --check
+
+  # Cross compile for multiple architectures.
+  #
+  # Cannot run the unit tests until we can specify a seccomp profile for cross'
+  # docker images. Since the default seccomp profile blocks keyctl calls.
+  libdbus-sys-linux:
+    runs-on: ubuntu-latest
+    needs: install-cross
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
+    - uses: dtolnay/rust-toolchain@stable
+      with:
+          toolchain: stable
+    - name: Download Cross
+      uses: actions/download-artifact@v3
+      with:
+        name: cross-linux-musl
+        path: /tmp
+    - run: rustup toolchain install ${{ matrix.channel }}
+    - run: chmod +x /tmp/cross
+    - name: Build
+      run: /tmp/cross build --package dbus --features vendored --target ${{ matrix.target }}
+    #- name: Run tests
+    #  run: /tmp/cross test --package libdbus-sys --features vendored --target ${{ matrix.target }}
+    strategy:
+      fail-fast: false
+      matrix:
+        channel: [stable]
+        target:
+          - mips64-unknown-linux-muslabi64
+          - mips-unknown-linux-musl
+          - aarch64-unknown-linux-musl
+          - arm-unknown-linux-musleabihf
+          - i686-unknown-linux-musl
+          - x86_64-unknown-linux-musl
+          - x86_64-unknown-linux-gnu
+
 
   # Run miri (commented out since miri doesn't support the keyctl
   # syscalls)

--- a/src/ffi/functions.rs
+++ b/src/ffi/functions.rs
@@ -67,7 +67,7 @@ pub(crate) fn keyctl_impl(
     arg3: libc::c_ulong,
     arg4: libc::c_ulong,
     arg5: libc::c_ulong,
-) -> Result<libc::c_ulong, KeyError> {
+) -> Result<libc::c_long, KeyError> {
     // Perform the actual system call
     let res = unsafe { libc::syscall(libc::SYS_keyctl, operation as u32, arg2, arg3, arg4, arg5) };
 
@@ -77,5 +77,5 @@ pub(crate) fn keyctl_impl(
     }
 
     // Otherwise return the result
-    res.try_into().or(Err(KeyError::InvalidIdentifier))
+    Ok(res)
 }

--- a/src/ffi/types.rs
+++ b/src/ffi/types.rs
@@ -180,11 +180,18 @@ impl From<KeySerialId> for i32 {
     }
 }
 
+/// Direct conversion
+impl From<i32> for KeySerialId {
+    fn from(n: i32) -> Self {
+        Self(n)
+    }
+}
+
 /// Allow easy conversion from u64 to KeySerialId
-impl TryFrom<u64> for KeySerialId {
+impl TryFrom<i64> for KeySerialId {
     type Error = KeyError;
 
-    fn try_from(n: u64) -> Result<Self, Self::Error> {
+    fn try_from(n: i64) -> Result<Self, Self::Error> {
         Ok(Self(n.try_into().or(Err(KeyError::InvalidIdentifier))?))
     }
 }

--- a/src/key.rs
+++ b/src/key.rs
@@ -235,6 +235,12 @@ mod tests {
     use zeroize::Zeroizing;
 
     #[test]
+    fn test_from_raw_id() {
+        let raw: i32 = 0x12345;
+        let _key = Key::from_id(raw.into());
+    }
+
+    #[test]
     fn test_metadata() {
         let secret = "Test Data";
 

--- a/src/keyring.rs
+++ b/src/keyring.rs
@@ -28,7 +28,7 @@ impl KeyRing {
         let id: KeySerialId = ffi::keyctl!(
             KeyCtlOperation::GetKeyRingId,
             id as libc::c_ulong,
-            u64::from(create)
+            u32::from(create).into()
         )?
         .try_into()
         .or(Err(KeyError::InvalidIdentifier))?;

